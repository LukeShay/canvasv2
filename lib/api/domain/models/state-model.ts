/* eslint-disable @typescript-eslint/no-var-requires */
/* eslint-disable global-require */
import { Model, JSONSchema } from 'objection';
import { IState } from '../../../types/state';
import Constants from '../constants';

export class StateModel extends Model implements IState {
  id: string;
  abbreviation: string;
  name: string;
  code: string;
  
  static tableName = Constants.STATES_TABLE;

  static idColumn = 'id';

  static get jsonSchema(): JSONSchema {
    return {
      properties: {
        abbreviation: { type: 'string' },
        code: { type: 'string' },
        name: { type: 'string' },
      },
      required: ['name', 'code', 'abbreviation'],
      type: 'object',
    };
  }
}
