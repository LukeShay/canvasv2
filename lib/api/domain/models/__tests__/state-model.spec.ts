import { StateModel } from '../state-model';

describe('StateModel', () => {
  it("should have idColum 'id'", () => {
    expect(StateModel.idColumn).toEqual('id');
  });

  it('should should have jsonSchema that matches snapshot', () => {
    expect(StateModel.jsonSchema).toMatchSnapshot();
  });

  it('should should have relationMappings that matches snapshot', () => {
    expect(StateModel.relationMappings).toMatchSnapshot();
  });
});
