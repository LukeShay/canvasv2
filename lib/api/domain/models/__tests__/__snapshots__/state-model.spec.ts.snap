// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateModel should should have jsonSchema that matches snapshot 1`] = `
Object {
  "properties": Object {
    "abbreviation": Object {
      "type": "string",
    },
    "code": Object {
      "type": "string",
    },
    "name": Object {
      "type": "string",
    },
  },
  "required": Array [
    "name",
    "code",
    "abbreviation",
  ],
  "type": "object",
}
`;

exports[`StateModel should should have relationMappings that matches snapshot 1`] = `null`;
